### PR TITLE
feat: added hmi extraction service to orchestration

### PR DESCRIPTION
**Pull request by tangopapatime**
# Description

Added the new `hmi-extraction-service` .  I think it's all here.

### CODE REVIEW

**Comment by tangopapatime** [comment]
@dgauldie @chris-dickson Let me know if this looks good.

File: kubernetes/base/services/hmi-extraction-service/hmi-extraction-service-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        software.uncharted.terarium/name: hmi-extraction-service
+    spec:
+      containers:
+        - name: hmi-extraction-service
+          image: hmi-extraction-service-image
+          env:
+            - name: RABBITMQ_DEFAULT_USER
+              value: terarium
+            - name: RABBITMQ_DEFAULT_PASS
+              value: terarium123
```
[the remark this answers is not in the document]
Comment (bigglesandginger) [reply kubernetes/base/services/hmi-extraction-service/hmi-extraction-service-deployment.yaml:30]: If these are not common to ALL deployments then they should not be in the base.  Look at hmi-server-deployment as an example - the base has no `env` section, where as the overlays do

File: kubernetes/base/services/hmi-extraction-service/hmi-extraction-service.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: hmi-extraction-service
+  labels:
+    software.uncharted.terarium/name: hmi-extraction-service
+    software.uncharted.terarium/component: extraction
+    software.uncharted.terarium/service: hmi-extraction
```
Comment (bigglesandginger) [inline kubernetes/base/services/hmi-extraction-service/hmi-extraction-service.yaml:9]: service no longer "desired" - delete line

File: kubernetes/base/services/hmi-extraction-service/hmi-extraction-service-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: hmi-extraction-service
+  labels:
+    software.uncharted.terarium/name: hmi-extraction-service
+    software.uncharted.terarium/component: extraction
+    software.uncharted.terarium/service: hmi-extraction
```
Comment (bigglesandginger) [inline kubernetes/base/services/hmi-extraction-service/hmi-extraction-service-deployment.yaml:9]: service no longer "desired" - delete line

File: kubernetes/overlays/dev/local/services/hmi-extraction-service/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - ../../../../../base/services/hmi-extraction-service
+patchesStrategicMerge:
+  - hmi-extraction-service.yaml
```
Comment (bigglesandginger) [inline kubernetes/overlays/dev/local/services/hmi-extraction-service/kustomization.yaml:7]: insert the patch for the env vars for `...deployment.yaml` here

File: kubernetes/overlays/prod/overlays/askem-staging/services/hmi-extraction-service/hmi-extraction-service.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+---
```
Comment (bigglesandginger) [inline kubernetes/overlays/prod/overlays/askem-staging/services/hmi-extraction-service/hmi-extraction-service.yaml:1]: This file does not appear to be referenced by an `kustomization.yaml` file.

**Review by bigglesandginger — CHANGES_REQUESTED** [review]
The staging might need more work to establish the env for the deployemnt, and with that secrets.  Obviously Production does not exist, but I assume this is "stage 1" get working iun dev and staging, then iron out the creases and create the production templates

File: kubernetes/base/services/hmi-extraction-service/hmi-extraction-service-deployment.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tangopapatime) [reply kubernetes/base/services/hmi-extraction-service/hmi-extraction-service-deployment.yaml:30]: this should be common to ALL deployments afaik just like the new message queue.

**Review by tangopapatime — COMMENTED** [review]
should be good to go.

File: kubernetes/overlays/dev/local/services/hmi-extraction-service/kustomization.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  - hmi-extraction-service-deployment.yaml
```
Comment (bigglesandginger) [inline kubernetes/overlays/dev/local/services/hmi-extraction-service/kustomization.yaml:8]: As you are not adding env vars - this line is not needed

**Review by bigglesandginger — APPROVED** [review]

**Pull request merged**